feat: 허브 검색 api 구현

AnJungGeon opened the pull request:
## 개요
허브 검색 api 구현하고 httpClient 테스트 진행했습니다

<br>

## 📝 작업 내용
### 변경 사항
필요 없는 import문 제거 했습니다.

@Setter 제거


### 변경 이유
Setter는 컨트롤러로 값을 전달받은 후 서비스 단에서 변경이 가능하므로 제거하였습니다.  
@ModelAttribute는 일반적으로 @Setter를 통해 자동 바인딩되지만,  
Setter가 없는 경우에는 바인딩이 불가능하기 때문에,  
WebDataBinder.initDirectFieldAccess()를 통해 필드 직접 접근 바인딩을 활성화하였습니다.
### 추가 설명

<br>

### 💬리뷰 요구사항

<br>

### #️⃣ 연관된 이슈
closed #188 

<br>

## PR 유형

이 PR은 어떤 변경 사항을 포함하고 있나요?

- [X] 새로운 기능 추가
- [ ] 버그 수정
- [ ] UI 디자인 변경 (CSS 등)
- [X] 코드에 영향을 주지 않는 변경사항 (오타 수정, 탭 사이즈 변경, 변수명 변경 등)
- [ ] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [X] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 또는 폴더명 수정
- [ ] 파일 또는 폴더 삭제
- [ ] 배포 준비, PR 관련 사항

<br>

## ✅ Check List

PR이 다음 요구 사항을 충족하는지 확인하세요.

- [X] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [X] 변경 사항에 대한 테스트를 했습니다. (버그 수정/기능에 대한 테스트)
- [ ] CI/CD 파이프라인을 통과했습니다.

AnJungGeon committed:
feat: 허브 QueryDsl 의존성 추가

허브 도메인 QueryDsl 의존성 추가했습니다

Resolved: #188

AnJungGeon committed:
del: 허브 Entity import 제거

허브 Entity 사용하지 않는 import문 제거 했습니다.

Resolved: #188

AnJungGeon committed:
feat: 허브 검색 요청 메소드 추가

허브 Controller 검색 요청 메소드 추가했습니다

Resolved: #188

AnJungGeon committed:
feat: 허브 검색 Repository 작성, 생성

허브 검색에 필요한 Repository 작성, 생성 했습니다

Resolved: #188

AnJungGeon committed:
feat: 허브 검색요청 request 생성

허브 검색에 필요한 request 생성 했습니다

Resolved: #188

AnJungGeon committed:
feat: 허브 검색 Mapper 생성

허브 검색 Mapper 생성했습니다

Resolved: #188

AnJungGeon committed:
feat: 허브 검색 비즈니스 로직 작성했습니다

허브 검색 비즈니스 로직 작성했습니다

Resolved: #188

AnJungGeon committed:
test: 허브 검색 httpClient 테스트

허브 검색 httpClient 테스트 했습니다

Resolved: #188

Jungdaye89 commented on hub/src/main/java/com/business/hub/application/dto/request/HubSearchRequest.java:
```diff
@@ -0,0 +1,12 @@
+import lombok.Setter;
+
+@Getter
+@Setter
```
DTO는 가능하면 불변 객체로 설계하는 것이 좋습니다.
setter를 사용하면 객체의 상태가 외부에서 변경될 수 있으므로, 생성자나 빌더 패턴을 활용하여 객체 생성 시 필요한 값들을 설정해 보면 좋을 것 같습니다!

AnJungGeon replied:
@ModelAttribute를 사용하려면 Setter가 필요해서 사용하였습니다.
다른 방법이 있는지 찾아보겠습니다

Jungdaye89 commented on hub/src/main/java/com/business/hub/application/mapper/HubMapper.java:
```diff
@@ -29,4 +30,12 @@ public static List<HubResponse> toDto(List<Hub> hubs) {
 
+    public static Page<HubResponse> toPageDto(Page<Hub> hubs) {
+        if (hubs == null || hubs.isEmpty()) {
+            throw new IllegalArgumentException("허브 데이터가 없습니다.");
```
에러 코드 컨벤션(BusinessException + ErrorCode enum 등)에 맞게 리팩토링이 필요합니다.

매퍼 쪽에서는 가능하면 검증보다는 변환 로직만 담당하고, 검증은 서비스에서 하는 것도 고려해보면 좋을 것 같아요 :)

AnJungGeon committed:
refector: 허브 Mapper 예외처리 수정

허브 Mapper 예외처리 수정했습니다

Resolved: #188

AnJungGeon committed:
refector: 허브 검색 요청 dto setter 제거

값이 변경 될 우려가 있어 setter 제거 했습니다

Resolved: #188

AnJungGeon committed:
feat: InitBinder 추가

Setter가 없으면 @ModelAttribute로 객체를 자동 바인딩이 불가하여 InitBinder로 가능하게 했습니다.
WebDataBinder.initDirectFieldAccess()로 필드 직접 접근 바인딩 활성화

Resolved: #188

AnJungGeon commented:
수정했습니다

dowon0113 commented:
수고하셨습니다
